core/src/session.rs

```diff
@@ -309,19 +309,17 @@ impl Session {
                 let mut user_attributes: UserAttributes = HashMap::new();
 
                 loop {
-                    match reader.read_event(&mut buf) {
+                    match reader.read_event_into(&mut buf) {
                         Ok(Event::Start(ref element)) => {
-                            current_element = std::str::from_utf8(element.name())?.to_owned()
+                            current_element = std::str::from_utf8(element)?.to_owned()
                         }
                         Ok(Event::End(_)) => {
                             current_element = String::new();
                         }
                         Ok(Event::Text(ref value)) => {
                             if !current_element.is_empty() {
-                                let _ = user_attributes.insert(
-                                    current_element.clone(),
-                                    value.unescape_and_decode(&reader)?,
-                                );
+                                let _ = user_attributes
+                                    .insert(current_element.clone(), value.unescape()?.to_string());
```

Comment:
Do we require `to_string`? If we bind to `_` i.e. drop te value anyway, we should not care what the `Cow` returns. Or it is a necessary compiler hint?

Reply:
Yes it is. See: https://github.com/librespot-org/librespot/actions/runs/4697151836/jobs/8327914727

```
error[E0308]: `?` operator has incompatible types
   --> core/src/session.rs:322:69
    |
322 | ...                   .insert(current_element.clone(),value.unescape()?);
    |                                                       ^^^^^^^^^^^^^^^^^- help: try using a conversion method: `.to_string()`
    |                                                       |
    |                                                       expected struct `std::string::String`, found enum `Cow`
    |
    = note: `?` operator cannot convert from `Cow<'_, str>` to `std::string::String`
    = note: expected struct `std::string::String`
                 found enum `Cow<'_, str>`

For more information about this error, try `rustc --explain E0308`.
error: could not compile `librespot-core` due to previous error
```